pkg/controller/tenant/tenant_controller_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+Copyright 2020 Authors of Arktos - file modified.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tenant
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	core "k8s.io/client-go/testing"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+func TestTenantCreation(t *testing.T) {
+
+	testcases := map[string]struct {
+		Tenant                  *v1.Tenant
+		ExpectCreatedNamespaces sets.String
+	}{
+		"new-tenants": {
+			Tenant: &v1.Tenant{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-tenant-1",
+				},
+			},
+			ExpectCreatedNamespaces: sets.NewString("default", "kube-system", "kube-public"),
```

Comment:
can we re-use the definition of tenant_default_namespaces created above? The namespace names should not be part of a test case config.

Reply:
Updated.